### PR TITLE
Feature/setup backend issue ticket 9

### PR DESCRIPTION
**Pull request by selibrah**
Ticket : [backend] add the logic to confirm a help, and display the help counts in each help post

- Established backend routes need [gethelps /gethelpbyid /updatehelp]

- implementing services for the routes.

- Add confirmation button

-Ensured users can confirm a need only once for accuracy.

### CODE REVIEW

File: src/app/api/help/[id]/service.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { updateFieldInDocument } from '../../../../utils/firebase/firestore/updateFieldInDocument.js';
+import getDocument from '../../../../utils/firebase/firestore/getDocument.js';
+import { arrayUnion } from "firebase/firestore";
+
+const updateFieldInHelp = updateFieldInDocument('helps');
+
+export async function getHelpById(id) {
+    const helpData = await getDocument('helps', id)
+    return { docId: helpData.result.id, ...helpData.result.data() };
```
Comment (soufyanelf) [inline src/app/api/help/[id]/service.js:9]: make sure the helpData exist. you can add ? 
or get result
 const { result } =

File: src/app/api/help/[id]/route.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+export async function PATCH(req, { params }) {
+    const { id } = params;
+    const { confirmation_count, ip } = JSON.parse(await req.text());
+    const result = await updateConfirmationCount(id, {
+        confirmation_count,
+        ip,
+    });
+    return result ? NextResponse.json({ success: true }) : NextResponse.json({ success: false })
```
Comment (soufyanelf) [inline src/app/api/help/[id]/route.js:17]: you can just return the result. but it's ok if you want to keep this line

File: src/app/helps/ConfirmButton/index.jsx [hunk 3]
```diff
@@ -1,27 +1,24 @@
...
 }) => {
-  const [isConfirmed, setIsConfirmed] = useState(false);
+  const isConfirmed = helpCount > 0 
   return (
     <Button
       style={{
         gap: 9,
       }}
-      onClick={() => {
-        // ?This is just an example
```
Comment (soufyanelf) [inline src/app/helps/ConfirmButton/index.jsx:17]: remove the comment

File: src/app/page.js [hunk 4]
```diff
@@ -1,48 +1,25 @@
+"use client";
```
Comment (soufyanelf) [inline src/app/page.js:1]: it's better to not use the client mode for the SEO

File: src/app/page.js [hunk 5]
```diff
@@ -1,48 +1,25 @@
...
-
- return data;
-}
-
-export default async function HelpsPage() {
-  const helps = await getHelps();
+export default  function HelpsPage() {
+  const [helps, setHelps] = useState([]);
+  const [ip, setIp] = useState("");
+  useEffect( () => {
```
Comment (soufyanelf) [inline src/app/page.js:13]: we don't need use effect to fetch the data

**Comment by soufyanelf** [comment]
@selibrah  I added couple comments  I would love to have @Everkers  review as well.

**Comment by Everkers** [comment]
image
You are leaking people's ip address ⚠️

**Comment by soufyanelf** [comment]
We can protect data in the prod project

**Comment by soufyanelf** [comment]
Otherwise is there other way to avoid the spam?

**Comment by Everkers** [comment]
> Otherwise is there other way to avoid the spam?

Relying on IP could be complex and might need constant maintenance. I believe starting with client-side cooldowns is a more straightforward approach. Perhaps in v2, we could allow a one-time confirmation from users; subsequent attempts would require authentication. This method would prevent spam while ensuring a smooth UX by not enforcing immediate authentication.

What are you thoughts on this @selibrah @soufyanelf

**Comment by soufyanelf** [comment]
Sounds good what about using local storage flag for this version?

**Comment by Everkers** [comment]
> Sounds good what about using local storage flag for this version?
> 
> @Everkers @selibrah

I think we should use localstorage along with cookies. They could mirror each other's values for redundancy. For instance, if the localStorage entry is removed but the cookie remains, we can repopulate the localStorage from the cookie, and vice versa. It would also be a +1 to use vague key names and, ideally, employ hashing for an additional security layer

**Comment by Everkers** [comment]
@selibrah , I'll be progressing on this in a separate branch. Please let me know your availability so we can keep you in the loop.

File: src/app/api/help/[id]/service.js (hunk 1 above)
Comment (selibrah) [reply src/app/api/help/[id]/service.js:9]: okey

**Comment by selibrah** [comment]
> @selibrah , I'll be progressing on this in a separate branch. Please let me know your availability so we can keep you in the loop.
 Thanks for the catch i will remove the ips from the result returned, but i think for now the ips is more useful

**Pull request closed without merging**